src/net_processing.cpp

```diff
@@ -1284,8 +1290,8 @@ void static ProcessGetData(CNode* pfrom, const CChainParams& chainparams, CConnm
             bool push = false;
             auto mi = mapRelay.find(inv.hash);
             int nSendFlags = (inv.type == MSG_TX ? SERIALIZE_TRANSACTION_NO_WITNESS : 0);
-            if (mi != mapRelay.end()) {
-                connman->PushMessage(pfrom, msgMaker.Make(nSendFlags, NetMsgType::TX, *mi->second));
+            if (mi != mapRelay.end() && mi->second.m_node_set.count(pfrom->GetId())) {
+                connman->PushMessage(pfrom, msgMaker.Make(nSendFlags, NetMsgType::TX, *mi->second.m_txref));
```

Comment:
I think a comment here would be helpful (since `mi->second.m_node_set.count(pfrom->GetId())` isn't immediately obvious when read outside the context of this PR). Something like:

```
// Only send the transaction if we've previously announced it to this peer
```